ui_testing/pages/order_page.py

```diff
@@ -179,13 +180,59 @@ def get_last_order_row(self):
     def get_shipment_date(self):
         return self.base_selenium.get_value(element='order:shipment_date')
 
-    def get_test_date(self):
-        return self.base_selenium.get_value(element='order:test_date')
+    def get_random_suborder_data(self, row_id=None):
+        
+        # get all the suborders
+        suborders = self.base_selenium.get_table_rows(element='order:suborder_table')
+        # select random row id
+        if not row_id:
+            if len(suborders) > 1:
+                row_id = randint(0, len(suborders) - 1)
+            else:
+                row_id = 0
+
+        self.info('Getting suborder no. {}'.format(row_id))
+
+        # get the current suborder data
+        suborder = self.base_selenium.get_row_cells_id_dict_related_to_header(
+            row=suborders[row_id], table_element='order:suborder_table')
+        # attach the row_id to the suborder
+        suborder['row_id'] = row_id
+        # attach the row element to the suborder
+        suborder['row_element'] = suborders[row_id]
+        return suborder
+
```

Comment:
you can use get_suborder_data and generate a random index and select its data from the return of the function instead of creating a new function

Reply:
@mazenmmbadr I don't need the suborder data, I need the random `row_id` selected and the the row element itself, which the `get_suborder_data` doesn't provide in return.